File: Implementation/Mathematics/BasicMath/python/padovanSequence.py
def isPS(arr, s):
    if s == 1:
        return True

    for i in range(3, s):
        if (arr[i] - arr[i - 2] != arr[i - 3]):
            return False

    return True

arr = [1, 1, 1, 0, 0, 0, 0, 0, 0, 0]
s = len(arr)
for i in range(3, s):
    arr[i] = arr[i - 2] + arr[i - 3]

if (isPS(arr, s)):
    print("Yes")
else:
    print("No")

File: Implementation/Mathematics/BasicMath/java/padovanSequence.java
import java.util.Arrays;

public class padovanSequence
{
    public static boolean isPS(int arr[], int s)
    {
        if (s == 1)
            return true;

        for (int i = 3; i < s; ++i)
            if (arr[i] - arr[i - 2] != arr[i - 3])
                return false;
        
        return true;
    }

    public static void main(String[] args)
    {
        int[] arr = new int[10];
        arr[0] = 1;
        arr[1] = 1;
        arr[2] = 1;
        int s = arr.length;
        Arrays.sort(arr);
        for (int i = 3; i < s; ++i)
            arr[i] = arr[i - 2] + arr[i - 3];

        if (isPS(arr, s))
            System.out.println("Yes");
        else
            System.out.println("No");
    }
}
